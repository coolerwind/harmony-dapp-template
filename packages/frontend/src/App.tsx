import React from 'react';
import styled from 'styled-components';
import { Unit } from '@harmony-js/utils'

import Account from './components/Account';
import Balance from './components/Balance';

import Logo from './img/harmony_logo.svg'
import { useHarmony } from './context/harmonyContext';

const App = () => {

  const hmy = useHarmony();

	return (
		<Wrapper>
      <Container>
        <Topbar>
          <img src={Logo} alt="Harmony logo" />
          <Flex>
            <Balance />
            <Account />
          </Flex>
        </Topbar>
        <Content>

        </Content>
      </Container>
		</Wrapper>
	);
};

const Flex = styled.div`
  display: flex;
  align-items: center;
`

const Topbar = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: 74px;
  width: 100%;
`;

const Content = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  flex: 1;
`

const Container = styled.div`
	display: flex;
	flex-direction: column;
	align-items: center;
	justify-content: center;
	min-height: 100vh;
  padding: 0px 24px;
  max-width: 1200px;
  margin: 0 auto;
`;

const Wrapper = styled.div`
	background-color: #0093E9;
  background-image: linear-gradient(160deg, #0093E9 0%, #80D0C7 100%);
	font-size: 1rem;
	color: white;
`;

export default App;
