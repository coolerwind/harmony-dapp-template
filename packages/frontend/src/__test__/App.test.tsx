import React from 'react';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';

import App from '../App';

describe('Test App component', () => {
	const component = <App />;

	const setup = () => render(component);

	it('renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(<App />, div);
		ReactDOM.unmountComponentAtNode(div);
	});

	it('Should match snapshot', () => {
		const { container } = setup();
		expect(container).toMatchSnapshot();
	});
});
