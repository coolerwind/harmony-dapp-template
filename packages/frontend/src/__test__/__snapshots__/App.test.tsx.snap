// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App component Should match snapshot 1`] = `
<div>
  <div
    class="sc-eCApnc boYvcd"
  >
    <div
      class="sc-hKFxyN gjUeMc"
    >
      <div
        class="sc-gtsrHT jNWTnQ"
      >
        <img
          alt="Harmony logo"
          src="harmony_logo.svg"
        />
        <div
          class="sc-bdnxRM cpQBWR"
        >
          <code
            aria-details="{}"
            aria-label="Balance"
          />
          <code
            aria-details="{}"
            aria-label="Account"
          />
        </div>
      </div>
      <div
        class="sc-dlnjwi hpYrtS"
      >
        <code
          aria-details="{}"
          aria-label="InfoContract"
        />
      </div>
    </div>
    <div
      class="Toastify"
    />
  </div>
</div>
`;
